test/MultisigTask.t.sol

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.15;
+
+import {IMulticall3} from "forge-std/interfaces/IMulticall3.sol";
+import {Test} from "forge-std/Test.sol";
+
+import {IGnosisSafe, Enum} from "@base-contracts/script/universal/IGnosisSafe.sol";
+
+import {MockTarget} from "test/mock/MockTarget.sol";
+import {MultisigTask} from "src/fps/task/MultisigTask.sol";
+import {AddressRegistry} from "src/fps/AddressRegistry.sol";
+import {MockMultisigTask} from "test/mock/MockMultisigTask.sol";
+
+contract MultisigTaskUnitTest is Test {
+    AddressRegistry public addresses;
+    MultisigTask public task;
+
+    string constant MAINNET_CONFIG = "./src/fps/example/task-03/mainnetConfig.toml";
+
+    function setUp() public {
+        vm.createSelectFork("mainnet");
+
+        // Instantiate the Addresses contract
+        addresses = new AddressRegistry(MAINNET_CONFIG);
+
+        // Instantiate the Mock MultisigTask contract
+        task = MultisigTask(new MockMultisigTask());
+    }
+
+    function testRunFailsNoNetworks() public {
+        vm.expectRevert("MultisigTask: no chains found");
+        task.run("./test/mock/invalidNetworkConfig.toml");
+    }
+
+    function testRunFailsEmptyActions() public {
+        /// add empty action that will cause a revert
+        MockMultisigTask(address(task)).addAction(address(0), "", 0, "");
+        vm.expectRevert("Invalid target for task");
+        task.run(MAINNET_CONFIG);
+    }
+
+    function testRunFailsInvalidAction() public {
+        /// add invalid args for action that will cause a revert
+        MockMultisigTask(address(task)).addAction(address(1), "", 0, "");
+        vm.expectRevert("Invalid arguments for task");
+        task.run(MAINNET_CONFIG);
+    }
+
+    function testBuildFailsAddressesNotSet() public {
+        vm.expectRevert("Must set addresses object for multisig address to be set");
+        task.build();
+    }
+
+    function testBuildFailsAddressesSetBuildStarted() public {
+        /// set multisig to a non zero address
+        vm.store(
+            address(task),
+            bytes32(uint256(38)),
+            bytes32(uint256(uint160(addresses.getAddress("SystemConfigOwner", getChain("optimism").chainId))))
+        );
+        /// set _buildStarted to true
+        vm.store(address(task), bytes32(uint256(52)), bytes32(uint256(1)));
```

Comment:
Similarly, can we explain these [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) too?